syntax = "proto3";

package bloock;

option go_package = "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto";
option java_package = "com.bloock.sdk.bridge.proto";

import "availability_entities.proto";
import "record_entities.proto";
import "shared.proto";
import "config.proto";


service AvailabilityService {
    rpc Publish(PublishRequest) returns (PublishResponse);
    rpc Retrieve(RetrieveRequest) returns (RetrieveResponse);
}

message PublishRequest {
    ConfigData config_data = 1;
    Record record = 2;
    Publisher publisher = 3;
}

message PublishResponse {
    string id = 1;
    optional Error error = 2;
}

message RetrieveRequest {
    ConfigData config_data = 1;
    Loader loader = 3;
}

message RetrieveResponse {
    Record record = 1;
    optional Error error = 2;
}